ci-bench/src/util/config.rs

```diff
@@ -0,0 +1,158 @@
+            ciphersuite,
+            version,
+            tag: tag.into(),
+        }
+    }
+}
+
+pub fn make_server_config(
+    params: &BenchmarkParam,
+    client_auth: ClientAuth,
+    resume: ResumptionParam,
+    max_fragment_size: Option<usize>,
+) -> ServerConfig<Ring> {
+    assert_eq!(params.ciphersuite.version(), params.version);
+    let client_auth = match client_auth {
+        ClientAuth::Yes => {
+            let roots = params.key_type.get_chain();
+            let mut client_auth_roots = RootCertStore::empty();
+            for root in roots {
+                client_auth_roots.add(&root).unwrap();
+            }
+            WebPkiClientVerifier::builder(Arc::new(client_auth_roots))
+                .build()
+                .unwrap()
+        }
+        ClientAuth::No => WebPkiClientVerifier::no_client_auth(),
+    };
+
+    let mut cfg = ServerConfig::builder()
+        .with_safe_default_cipher_suites()
+        .with_safe_default_kx_groups()
+        .with_protocol_versions(&[params.version])
+        .unwrap()
+        .with_client_cert_verifier(client_auth)
+        .with_single_cert(params.key_type.get_chain(), params.key_type.get_key())
+        .expect("bad certs/private key?");
+
+    if resume == ResumptionParam::SessionID {
+        cfg.session_storage = ServerSessionMemoryCache::new(128);
+    } else if resume == ResumptionParam::Tickets {
+        cfg.ticketer = Ticketer::new().unwrap();
+    } else {
+        cfg.session_storage = Arc::new(NoServerSessionStorage {});
+    }
+
+    cfg.max_fragment_size = max_fragment_size;
+    cfg
+}
+
+pub fn make_client_config(
+    params: &BenchmarkParam,
+    clientauth: ClientAuth,
+    resume: ResumptionParam,
+) -> ClientConfig<Ring> {
+    assert_eq!(params.ciphersuite.version(), params.version);
+    let mut root_store = RootCertStore::empty();
+    let mut rootbuf =
+        io::BufReader::new(fs::File::open(params.key_type.path_for("ca.cert")).unwrap());
+    root_store.add_parsable_certificates(&rustls_pemfile::certs(&mut rootbuf).unwrap());
+
+    let cfg = ClientConfig::builder()
+        .with_cipher_suites(&[params.ciphersuite])
+        .with_safe_default_kx_groups()
+        .with_protocol_versions(&[params.version])
+        .unwrap()
+        .with_root_certificates(root_store);
+
+    let mut cfg = if clientauth == ClientAuth::Yes {
+        cfg.with_client_auth_cert(
+            params.key_type.get_client_chain(),
+            params.key_type.get_client_key(),
+        )
+        .unwrap()
+    } else {
+        cfg.with_no_client_auth()
+    };
+
+    if resume != ResumptionParam::No {
+        cfg.resumption = Resumption::in_memory_sessions(128);
+    } else {
+        cfg.resumption = Resumption::disabled();
+    }
+
+    cfg
+}
+
+pub fn new_client(
+    params: &BenchmarkParam,
+    clientauth: ClientAuth,
+    resume: ResumptionParam,
+) -> ClientConnection {
+    let client_config = Arc::new(make_client_config(params, clientauth, resume));
```

Comment:
I would prefer inlining `make_client_config()` and `make_server_config()` into these functions.